test/integration/database-helper_test.js

```diff
@@ -114,29 +113,31 @@ describe('Integration | db-connection.js', () => {
       expect(skillIdsCount).to.deep.equal(skillCount);
     });
 
-    it('should throw an error if the data to be inserted has no primary key', async function() {
-      const table = {
-        name: 'challenges',
-        fields: [
-          { name: 'instructions', type: 'text' },
-          { name: 'timer', type: 'smallint' },
-          { name: 'autoReply', type: 'boolean' },
-          { name: 'skillIds', type: 'text []', isArray: true },
-          { name: 'skillCount', type: 'smallint', extractor: (record) => _.size(record['skillIds']) },
-          { name: 'firstSkillId', type: 'text', extractor: (record) => _.get(record['skillIds'], 0) },
-        ],
-        indices: ['firstSkillId'],
-      };
-      const fullLearningContent = {
-        challenges: [{}],
-      };
-      const learningContent = fullLearningContent[table.name];
+    context('when at least one of the collection item has no property', () => {
+      it('should throw an error', async function() {
+        const table = {
+          name: 'challenges',
+          fields: [
+            { name: 'instructions', type: 'text' },
+            { name: 'timer', type: 'smallint' },
+            { name: 'autoReply', type: 'boolean' },
+            { name: 'skillIds', type: 'text []', isArray: true },
+            { name: 'skillCount', type: 'smallint', extractor: (record) => _.size(record['skillIds']) },
+            { name: 'firstSkillId', type: 'text', extractor: (record) => _.get(record['skillIds'], 0) },
+          ],
+          indices: ['firstSkillId'],
+        };
+        const fullLearningContent = {
+          challenges: [{}],
+        };
+        const learningContent = fullLearningContent[table.name];
+
+        const error = await catchErr(databaseHelper.saveLearningContent)(table, learningContent, databaseConfig);
+
+        expect(error.message).to.equal('At least one item to be inserted in table challenges has no property at all');
+      });
 
-      const error = await catchErr(databaseHelper.saveLearningContent)(table, learningContent, databaseConfig);
-
-      expect(error).to.be.instanceof(PrimaryKeyNotNullConstraintError);
```

Comment:
J'aimais bien l'idée qu'on teste le type d'erreur plutot que son message. Si on veut tester le message, on peut tester la classe d'erreur en question.

Reply:
Corrigé ! J'ai repris le custom helper chai du monorepo 
`deepEqualInstanceOmitting`

Reply:
On a du coup une méga assertion alors que `expect(error).to.be.instanceof(PrimaryKeyNotNullConstraintError);` c'était bien ça.

Reply:
Très juste ! J'ai remis la version d'origine.